src/pages/index.js

```diff
@@ -6,121 +6,415 @@ import Section from "../components/Section.js";
+  // profileDescriptionInput.value = userInfo.about; 
+    
+    const name = profileTitleInput.value;
+    const about = profileDescriptionInput.value;
+
+    api
+    .editProfile({name, about})
+    .then((result) => {
+      console.log(result);
+      editUserInfo.setUserInfo({name: result.name, about: result.about});
+      editProfileSubmitButton.textContent = "Saving..."
+      editProfilePopup.close();
+    })
+    .catch((err) => {
+      console.error(err);
+      console.error("Changes to profile submission unsuccessful. Error."); 
+    })
+    .finally(() => {
+      editProfilePopup.submitButtonLoadingState(true); 
+      
+    });
+  }
+  
+
+  editProfileFormValidator.enableValidation();
+  
+  
+
+
+  function handleProfileUpdateAvatarFormSubmit() {
+
+    updateAvatarPopup.submitButtonLoadingState(false);
+
+    const avatar = profileAvatarUrlInput.value;
+
+    api.editProfileAvatar({avatar})
+    .then((result) => {
+      console.log(result);
+      editUserInfo.setUserAvatarInfo({avatar: result.avatar});
+      updateAvatarPopup.close();
+  console.log("Avatar has been changed to the following avatar", avatar)
+    })
+    .catch((err) => {
+      console.error(err);
+      console.error("Profile Avatar edit api unsuccessful. Error")
+    }) 
+    .finally(() => {
+      updateAvatarPopup.submitButtonLoadingState(true); 
+    });
+  }
+
+
+
+
+//CARDS
+let cardSection;
+
+api
+  .getInitialCards()
+  .then((result) => {
+    console.log(result);
+    cardSection = new Section(
+      {
+        items: result,
+        renderer: createCard,
+      },
+      ".cards__list"
+    );
+    cardSection.renderItems();
+    return result;
+  })
+  .catch((err) => {
+    console.error(err); 
+    console.error("Card retrieval unsuccessful. Error");
+  });
+
+
+// TODO call api.getUserInfo
+// in the .then(), set the user info
+
 // FUNCTIONS
 
+
 function handleImageClick(cardData) {
   modalImage.alt = cardData.name;
   modalImage.src = cardData.link;
   modalText.textContent = cardData.name;
   cardPreviewPopup.open(cardData);
+  console.log(cardData);
 }
 
-function createCard(cardData) {
-  const card = new Card(cardData, "#card-template", handleImageClick);
 
+function handleConfirmDeleteSubmit(cardId, cardElement) {
+;
+  api
+  .deleteCard(cardId, cardElement)
+  .then(() => {
+    cardElement.remove(cardId);
+    console.log(cardId, "post has been deleted");
+    cardElement = null;
+    deleteCardPopup.close();
+  })
+  .catch((err) => {
+    console.error(err);
+    console.error("Card deletion unsuccessful. Error")
+  });
+}
+
+function handleDeletePopup(cardId, cardElement) { 
+deleteCardPopup.open(cardId, cardElement);
+}
+
+
+function checkLikeStatus(isLiked, cardData) {
+  
+    if (!isLiked) {
+      this._likeButton.classList.remove("cards__like-button_active");
+    } else {
+      (this._likeButton.classList.add("cards__like-button_active")); 
+      console.log("ANOTHER TOGGLE VIA CHECKSTATUS");
+    }
```

Comment:
You don't need it in `index.js` because you can handle it inside `Card`

Now only class `Card` knows every **card selector**, so you need to encapsulate the likes logics inside its methods and call them in `index.js`. It’s a violation of **Single responsibility** principle